### ISSUE TITLE
Implement XMLHttpRequest.responseURL

### ISSUE THREAD

**Issue by jdm** [servo/servo#8830]
We have a method that ostensibly implements the specification (`XMLHttpRequest::ResponseURL`), but we don't actually set the `response_url` member of `XMLHttpRequest` anywhere except the constructor. We should set this value appropriately inside `process_partial_response` in the `XHRProgress::Done` branch, and the member will need to be `DOMRefCell<DOMString>` (rather than a plain `DOMString`) to accommodate the new mutation.

Code: `components/script/dom/xmlhttprequest.rs`
Tests: `./mach test-wpt tests/wpt/web-platform-tests/XMLHttpRequest`

**Comment by jackinloadup** [comment]
I'd like to try and take this task on

**Comment by KiChjang** [comment]
You got it!

**Comment by jdm** [comment]
Oh, I forgot to include the specification in the original description: https://xhr.spec.whatwg.org/#the-responseurl-attribute

**Comment by KiChjang** [comment]
@jackinloadup any updates on this?

**Comment by jackinloadup** [comment]
I have some code in place but I have been having some difficulty figuring out how to debug the internals of various values.

Namely how to use gdb or some other tool to inspect aspects of hyper headers or url fields.

https://github.com/jackinloadup/servo/commit/be1d04f3ba5b479c33f88330e0972334f2ab6c1e

**Comment by jdm** [comment]
@jackinloadup There are `rust-lldb` and `rust-gdb` wrappers that include pretty-printers in `.servo/rust/[hash]/rustc-1.0.0-[platform]/bin/` that can be run against the `target/debug/servo` binary.

**Comment by jdm** [comment]
I use regular lldb on OS X for the same purpose, too.

**Comment by KiChjang** [comment]
@jackinloadup still working on this issue?

**Comment by shinglyu** [comment]
May I give it a try?

**Comment by KiChjang** [comment]
Yes, absolutely!

**Comment by shinglyu** [comment]
I tried to write a wpt test to cover this. But I was unable to run the single test file.
STR:
* Use `./mach create-wpt tests/wpt/web-platform-tests/XMLHttpRequest/responseurl.htm`
* Copy the `responsexml-basic.htm` test into `responseurl.htm`
* Run with `./mach test-wpt tests/wpt/web-platform-tests/XMLHttpRequest/responseurl.htm`

Expected:
The test runs

Actual:
The test starts, opens a blak Servo window and hangs forever

Notes:
I tried `./mach test-wpt tests/wpt/web-platform-tests/XMLHttpRequest` and it works, no Servo window poping up, and the test can catch errors if I add an assert(false) in it. But running a single file (I tried `./mach test-wpt tests/wpt/web-platform-tests/XMLHttpRequest/responsexml-basic.htm`) it just doesn't run. What's the difference between running a single file and a folder?
